File: Minimum Number of Arrows to Burst Balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int count = 1, i = 0, j = 1, n = points.size();
        int ps = points[0][0];
        int pe = points[0][1];
        while (j < n) {
            int s = points[j][0];
            int e = points[j][1];
            if (pe < s) {
                count++;
                ps = s;
                pe = e;
                j++;
            } else {
                if (s > ps)
                    ps = s;
                if (e < pe)
                    pe = e;
                j++;
            }
        }
        return count;
    }
};

File: Minimum Number of Arrows to Burst Balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution(object):
    def findMinArrowShots(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
       
        points.sort()

        count = 1  
        n = len(points)
        ps, pe = points[0][0], points[0][1]  

        j = 1
        while j < n:
            s, e = points[j][0], points[j][1]  
            
            if pe < s:
                
                count += 1
                ps, pe = s, e
            else:
               
                ps = max(ps, s)
                pe = min(pe, e)
            
            j += 1

        return count
